src/Form/index.tsx

```diff
@@ -104,6 +104,43 @@ import usePrevious from '../hooks/usePrevious';
 import ReactPortal from './components/ReactPortal';
 import { replaceTextVariables } from '../elements/components/TextNodes';
 
+// getFormContext needs to live outside of the component so that we can export
+// the types for the end user
+const getFormContext = (
+  newStep: any,
+  props: {
+    history: any;
+    client: any;
+    updateFieldOptions: (
+      stepData: any,
+      loadStep?: null
+    ) => (newOptions: FieldOptions) => void;
+    setUserProgress: React.Dispatch<React.SetStateAction<null>>;
+    steps: any;
+    activeStep: any;
+  }
+) => ({
+  setValues,
+  setFormCompletion: (flag: boolean) =>
+    props.client.registerEvent({
+      step_key: props.activeStep.key,
+      event: 'load',
+      completed: flag
+    }),
+  setOptions: props.updateFieldOptions(props.steps),
+  setProgress: (val: any) => props.setUserProgress(val),
+  setStep: (stepKey: any) => {
+    changeStep(stepKey, newStep.key, props.steps, props.history);
+  },
+  step: {
```

Comment:
attributes: { stepBackgroundColor: newStep?.default_background_color } maybe is cleaner?